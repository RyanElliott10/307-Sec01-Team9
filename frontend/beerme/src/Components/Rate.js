import React, { Component } from 'react'
import StarRatings from 'react-star-ratings';
import Photo from "../img/BOTD_photo.png";
import * as Constants from "..//Utils/Constants";
import { Form } from "react-bootstrap";
import { Redirect, Link } from "react-router-dom";

import UserController from "../Controllers/UserController";
import NetClient from '../Utils/NetClient';

export default class Rate extends Component {
    constructor(props) {
        super(props);
        this.state = {};
    }

    componentDidMount() {
      NetClient.get(`https://localhost:44300/api/beerratings/${UserController.currBeerId}`).then(data => {
        console.log(data);
        this.setState({
          currentBeerRating: data
        })
      });
    }

    changeRating = (newRating, name ) => {
      this.setState({
        rating: newRating
      });

      const rateData = {
        UserId: UserController.userId,
        BeerId: UserController.currBeerId,
        Rating: newRating
      };

      NetClient.post("https://localhost:44300/api/beerratings", rateData).then(data => console.log(data));
    }

    renderBody() {
      return(
        <div style={styles.inColStyle}>
            <div style={styles.inRowStyle}>
              <img
                src={Photo}
                alt="Photo"
                style={{ width: 110, height: 180, marginRight: "20px" }}
              />
              <div style={styles.inBodyStyle}>    
                <h5>
                  {`Style: ${UserController.currStyle}`}
                </h5>
                <h5>
                  {`Alcohol Per Volume: ${UserController.currABV}`}
                </h5>
                <h5>
                  {`International Bittering Unit: ${UserController.currIBU}`}
                </h5>
              </div>   
            </div>
          </div>
      )
    }

    renderStars() {
      if (!UserController.isLoggedIn) {
        return(
          <Form style= {{alignItems: "center", justifyContent: "center"}}>
            <Form.Row>
            <Form.Label>{"In order to rate this beer, please"}</Form.Label>
              <Link
                to="/account-entry"
                style={{ color: "blue", marginLeft: "5px" }}
              >
                create an account.
              </Link>
            </Form.Row>
          </Form>
        )
      }
      else return (
        <div style={styles.inColStyle}>
          <h5>
            Rate this beer!
          </h5>
          <StarRatings
            rating={this.state.rating}
            starRatedColor= {Constants.ORANGE_COLOR}
            starHoverColor="grey"
            changeRating={this.changeRating}
            numberOfStars={5}
            name='rating'
            starSpacing="15px"
          />
        </div>
      )
    }

   
    render() {
      // rating = 2;
      return (
        <div style={styles.inTitleStyle}>
          <h1>
            {UserController.currBeer}
          </h1>
          <h5 style={{alignItems: "right"}}>
            {`Average Rating: ${this.state.currentBeerRating ? this.state.currentBeerRating.toFixed(2) : "Not yet rated"}`}
          </h5>
          {this.renderBody()}
          {this.renderStars()}
          
        </div>
      );
    }
}

const styles = {
  inColStyle: {
    display: "flex",
    flexDirection: "column",
    justifyContent: "space-around",
    marginRight: "30px", 
    alignItems: "center"
  },
  inBodyStyle: {
    display: "flex",
    flexDirection: "column",
    justifyContent: "space-around",
    marginLeft: "30px", 
    marginTop: "60px",
    marginBottom: "90px",
    alignItems: "left"   
  },
  inTitleStyle: {
    display: "flex",
    flexDirection: "column",
    marginLeft: "200px",
    marginRight: "200px", 
    marginBottom: "100px"
  },
  inRowStyle: {
    display: "flex",
    flexDirection: "row",
    marginTop: "50px"
  }
};

class Bar extends Component {
    render() {
      // aggregateRating = 2.35;
      return (
        <StarRatings
          rating={2.403}
          starDimension="40px"
          starSpacing="15px"
        />
      );
    }
  }