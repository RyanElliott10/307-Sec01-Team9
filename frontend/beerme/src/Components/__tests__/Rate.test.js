import React from "react";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";

import Rate from "../Rate";

it("renders correctly", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <Rate />
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
