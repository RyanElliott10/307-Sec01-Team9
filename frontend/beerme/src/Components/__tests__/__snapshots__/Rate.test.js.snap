// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "flexDirection": "column",
      "marginBottom": "100px",
      "marginLeft": "200px",
      "marginRight": "200px",
    }
  }
>
  <h1>
    
  </h1>
  <h5
    style={
      Object {
        "alignItems": "right",
      }
    }
  >
    Average Rating: Not yet rated
  </h5>
  <div
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "column",
        "justifyContent": "space-around",
        "marginRight": "30px",
      }
    }
  >
    <div
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
          "marginTop": "50px",
        }
      }
    >
      <img
        alt="Photo"
        src="BOTD_photo.png"
        style={
          Object {
            "height": 180,
            "marginRight": "20px",
            "width": 110,
          }
        }
      />
      <div
        style={
          Object {
            "alignItems": "left",
            "display": "flex",
            "flexDirection": "column",
            "justifyContent": "space-around",
            "marginBottom": "90px",
            "marginLeft": "30px",
            "marginTop": "60px",
          }
        }
      >
        <h5>
          Style: 
        </h5>
        <h5>
          Alcohol Per Volume: 
        </h5>
        <h5>
          International Bittering Unit: 
        </h5>
      </div>
    </div>
  </div>
  <form
    className=""
  >
    <div
      className="form-row"
      style={
        Object {
          "alignItems": "top",
          "justifyContent": "center",
        }
      }
    >
      <label
        className="form-label"
      >
        Please 
      </label>
      <a
        href="/account-entry"
        onClick={[Function]}
        style={
          Object {
            "color": "blue",
            "marginLeft": "5px",
            "marginRight": "5px",
          }
        }
      >
        create an account
      </a>
      <label
        className="form-label"
      >
         to rate this beer.
      </label>
    </div>
  </form>
</div>
`;

exports[`renders correctly for businesses 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "flexDirection": "column",
      "marginBottom": "100px",
      "marginLeft": "200px",
      "marginRight": "200px",
    }
  }
>
  <h1>
    
  </h1>
  <h5
    style={
      Object {
        "alignItems": "right",
      }
    }
  >
    Average Rating: Not yet rated
  </h5>
  <div
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "column",
        "justifyContent": "space-around",
        "marginRight": "30px",
      }
    }
  >
    <div
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
          "marginTop": "50px",
        }
      }
    >
      <img
        alt="Photo"
        src="BOTD_photo.png"
        style={
          Object {
            "height": 180,
            "marginRight": "20px",
            "width": 110,
          }
        }
      />
      <div
        style={
          Object {
            "alignItems": "left",
            "display": "flex",
            "flexDirection": "column",
            "justifyContent": "space-around",
            "marginBottom": "90px",
            "marginLeft": "30px",
            "marginTop": "60px",
          }
        }
      >
        <h5>
          Style: 
        </h5>
        <h5>
          Alcohol Per Volume: 
        </h5>
        <h5>
          International Bittering Unit: 
        </h5>
      </div>
    </div>
  </div>
  <form
    className=""
  >
    <div
      className="form-row"
      style={
        Object {
          "alignItems": "top",
          "justifyContent": "center",
          "marginTop": "50px",
        }
      }
    >
      <label
        className="form-label"
      >
        As a business, you may not rate this beer.
      </label>
    </div>
  </form>
</div>
`;

exports[`renders correctly for logged in user 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "flexDirection": "column",
      "marginBottom": "100px",
      "marginLeft": "200px",
      "marginRight": "200px",
    }
  }
>
  <h1>
    
  </h1>
  <h5
    style={
      Object {
        "alignItems": "right",
      }
    }
  >
    Average Rating: Not yet rated
  </h5>
  <div
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "column",
        "justifyContent": "space-around",
        "marginRight": "30px",
      }
    }
  >
    <div
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
          "marginTop": "50px",
        }
      }
    >
      <img
        alt="Photo"
        src="BOTD_photo.png"
        style={
          Object {
            "height": 180,
            "marginRight": "20px",
            "width": 110,
          }
        }
      />
      <div
        style={
          Object {
            "alignItems": "left",
            "display": "flex",
            "flexDirection": "column",
            "justifyContent": "space-around",
            "marginBottom": "90px",
            "marginLeft": "30px",
            "marginTop": "60px",
          }
        }
      >
        <h5>
          Style: 
        </h5>
        <h5>
          Alcohol Per Volume: 
        </h5>
        <h5>
          International Bittering Unit: 
        </h5>
      </div>
    </div>
  </div>
  <div
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "column",
        "justifyContent": "space-around",
        "marginRight": "30px",
      }
    }
  >
    <h5>
      Rate this beer!
    </h5>
    <div
      className="star-ratings"
      style={
        Object {
          "boxSizing": "border-box",
          "display": "inline-block",
          "position": "relative",
        }
      }
      title="0 Stars"
    >
      <svg
        className="star-grad"
        style={
          Object {
            "height": "0",
            "position": "absolute",
            "visibility": "hidden",
            "width": "0",
            "zIndex": "0",
          }
        }
      >
        <defs>
          <linearGradient
            id="starGrad789014827622014"
            x1="0%"
            x2="100%"
            y1="0%"
            y2="0%"
          >
            <stop
              className="stop-color-first"
              offset="0%"
              style={
                Object {
                  "stopColor": "#CC752A",
                  "stopOpacity": "1",
                }
              }
            />
            <stop
              className="stop-color-first"
              offset="0%"
              style={
                Object {
                  "stopColor": "#CC752A",
                  "stopOpacity": "1",
                }
              }
            />
            <stop
              className="stop-color-final"
              offset="0%"
              style={
                Object {
                  "stopColor": "rgb(203, 211, 227)",
                  "stopOpacity": "1",
                }
              }
            />
            <stop
              className="stop-color-final"
              offset="100%"
              style={
                Object {
                  "stopColor": "rgb(203, 211, 227)",
                  "stopOpacity": "1",
                }
              }
            />
          </linearGradient>
        </defs>
      </svg>
      <div
        className="star-container"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "cursor": "pointer",
            "display": "inline-block",
            "paddingLeft": undefined,
            "paddingRight": "15px",
            "position": "relative",
            "verticalAlign": "middle",
          }
        }
      >
        <svg
          className="widget-svg"
          style={
            Object {
              "height": "50px",
              "transform": undefined,
              "transition": "transform .2s ease-in-out",
              "width": "50px",
            }
          }
          viewBox="0 0 51 48"
        >
          <path
            className="star"
            d="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
            style={
              Object {
                "fill": "rgb(203, 211, 227)",
                "transition": "fill .2s ease-in-out",
              }
            }
          />
        </svg>
      </div>
      <div
        className="star-container"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "cursor": "pointer",
            "display": "inline-block",
            "paddingLeft": "15px",
            "paddingRight": "15px",
            "position": "relative",
            "verticalAlign": "middle",
          }
        }
      >
        <svg
          className="widget-svg"
          style={
            Object {
              "height": "50px",
              "transform": undefined,
              "transition": "transform .2s ease-in-out",
              "width": "50px",
            }
          }
          viewBox="0 0 51 48"
        >
          <path
            className="star"
            d="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
            style={
              Object {
                "fill": "rgb(203, 211, 227)",
                "transition": "fill .2s ease-in-out",
              }
            }
          />
        </svg>
      </div>
      <div
        className="star-container"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "cursor": "pointer",
            "display": "inline-block",
            "paddingLeft": "15px",
            "paddingRight": "15px",
            "position": "relative",
            "verticalAlign": "middle",
          }
        }
      >
        <svg
          className="widget-svg"
          style={
            Object {
              "height": "50px",
              "transform": undefined,
              "transition": "transform .2s ease-in-out",
              "width": "50px",
            }
          }
          viewBox="0 0 51 48"
        >
          <path
            className="star"
            d="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
            style={
              Object {
                "fill": "rgb(203, 211, 227)",
                "transition": "fill .2s ease-in-out",
              }
            }
          />
        </svg>
      </div>
      <div
        className="star-container"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "cursor": "pointer",
            "display": "inline-block",
            "paddingLeft": "15px",
            "paddingRight": "15px",
            "position": "relative",
            "verticalAlign": "middle",
          }
        }
      >
        <svg
          className="widget-svg"
          style={
            Object {
              "height": "50px",
              "transform": undefined,
              "transition": "transform .2s ease-in-out",
              "width": "50px",
            }
          }
          viewBox="0 0 51 48"
        >
          <path
            className="star"
            d="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
            style={
              Object {
                "fill": "rgb(203, 211, 227)",
                "transition": "fill .2s ease-in-out",
              }
            }
          />
        </svg>
      </div>
      <div
        className="star-container"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "cursor": "pointer",
            "display": "inline-block",
            "paddingLeft": "15px",
            "paddingRight": undefined,
            "position": "relative",
            "verticalAlign": "middle",
          }
        }
      >
        <svg
          className="widget-svg"
          style={
            Object {
              "height": "50px",
              "transform": undefined,
              "transition": "transform .2s ease-in-out",
              "width": "50px",
            }
          }
          viewBox="0 0 51 48"
        >
          <path
            className="star"
            d="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
            style={
              Object {
                "fill": "rgb(203, 211, 227)",
                "transition": "fill .2s ease-in-out",
              }
            }
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
